### ISSUE TITLE
Error in compiling python wrapper with MPI

### ISSUE THREAD

**Issue by akuesters** [arbor-sim/arbor#594]
compiling python wrapper with MPI

```
export CC=`which mpicc`
export CXX=`which mpicxx`
cmake -DARB_WITH_MPI=ON -DARB_WITH_PYTHON=ON
```

will get error of type `incomplete type 'ompi_communicator_t'`

```
[ 95%] Building CXX object python/CMakeFiles/pyarb.dir/context.cpp.o
In file included from /Users/kuesters/SynologyDrive/Work/97_FZJ/98_JSC/Coding/Arbor/GitHub/akuesters_arbor/python/context.cpp:1:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:169:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:640:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:1361:59: error: incomplete type 'ompi_communicator_t' used
      in type trait expression
    : public integral_constant<bool, __is_base_of(_Bp, _Dp)> {};
                                                          ^
```

**Comment by akuesters** [comment]
source: 

```
/Users/kuesters/SynologyDrive/Work/97_FZJ/98_JSC/Coding/Arbor/GitHub/akuesters_arbor/python/context.cpp:54:10: note: in instantiation of function template specialization
      'pybind11::class_<pyarb::context_shim>::def<(lambda at
      /Users/kuesters/SynologyDrive/Work/97_FZJ/98_JSC/Coding/Arbor/GitHub/akuesters_arbor/python/context.cpp:55:13), pybind11::detail::void_type (*)(),
      pyarb::context_shim (const arb::proc_allocation &, ompi_communicator_t *), pybind11::detail::void_type ()>' requested here
        .def(pybind11::init(
         ^
/opt/local/include/openmpi-mp/mpi.h:330:16: note: forward declaration of 'ompi_communicator_t'
typedef struct ompi_communicator_t *MPI_Comm;
```

**Comment by bcumming** [comment]
This was fixed in commit 1d6a48d0ce4b96f59acf931efd61d55c571c5e68
